execution/src/execution.rs

```diff
@@ -262,6 +262,45 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
 
         Ok(Some(tx))
     }
+
+    pub async fn get_logs(
+        &self,
+        filter: &Filter,
+        payloads: &BTreeMap<u64, ExecutionPayload>,
+    ) -> Result<Vec<Log>> {
+        let logs = self.rpc.get_logs(filter).await?;
+
+        for (_pos, log) in logs.iter().enumerate() {
+            // For every log
+            // Get the hash of the tx that generated it
+            let tx_hash = log.transaction_hash.unwrap();
```

Comment:
Lets avoid doing a naked `unwrap` to prevent it from panicing. Take a look at the `ok_or` function method which converts options to results.